File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/Video/CustomMediaController.java
package com.kyleriedemann.giantbombvideoplayer.Video;

import android.app.Activity;
import android.content.Context;
import android.util.AttributeSet;
import android.view.KeyEvent;
import android.widget.MediaController;

/**
 * Created by kyle on 3/17/16.
 */
public class CustomMediaController extends MediaController {
    public CustomMediaController(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomMediaController(Context context, boolean useFastForward) {
        super(context, useFastForward);
    }

    public CustomMediaController(Context context) {
        super(context);
    }

    @Override
    public boolean dispatchKeyEvent(KeyEvent event)
    {
        if (event.getKeyCode() == KeyEvent.KEYCODE_BACK)
            ((Activity) getContext()).finish();

        return super.dispatchKeyEvent(event);
    }
}

File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/video/Player/CustomMediaController.kt
package com.kyleriedemann.giantbombvideoplayer.video.player

import android.app.Activity
import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent
import android.widget.MediaController

class CustomMediaController : MediaController {
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, useFastForward: Boolean) : super(context, useFastForward) {}

    constructor(context: Context) : super(context) {}

    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
        if (event.keyCode == KeyEvent.KEYCODE_BACK)
            (context as Activity).finish()

        return super.dispatchKeyEvent(event)
    }
}
